kaggle_environments/envs/werewolf/eval/metrics.py

```diff
@@ -0,0 +1,875 @@
+        self.openskill_mu_std: float = 0.0
+
+    def set_agent_name(self, name: str):
+        self.agent_name = name
+        self.openskill_rating = self.openskill_model.rating(name=name) if self.openskill_model else None
+
+    def get_win_rate(self) -> Tuple[float, float]:
+        return _mean_sem(self.wins)
+
+    def get_win_rate_for_role(self, role: str) -> Tuple[float, float]:
+        return _mean_sem(self.wins_by_role.get(role, []))
+
+    def get_irp(self) -> Tuple[float, float]:
+        return _mean_sem(self.irp_scores)
+
+    def get_vss(self) -> Tuple[float, float]:
+        return _mean_sem(self.vss_scores)
+
+    def get_ksr(self) -> Tuple[float, float]:
+        return _mean_sem(self.survival_scores)
+
+    def get_ksr_for_role(self, role: str) -> Tuple[float, float]:
+        return _mean_sem(self.survival_by_role.get(role, []))
+
+
+class GameSetEvaluator:
+    """Evaluates a set of game replays and calculates metrics for each agent."""
+
+    def __init__(self, input_dir: Union[str, List[str]], gte_tasks: List[str] = None,
+                 preserve_full_game_records: bool = False):
+        if isinstance(input_dir, str):
+            input_dirs = [input_dir]
+        else:
+            input_dirs = input_dir
+
+        self.games = []
+        for directory in input_dirs:
+            self.games.extend(get_game_results(directory, preserve_full_record=preserve_full_game_records))
+
+        self.openskill_model = PlackettLuce() if OPENSKILL_AVAILABLE else None
+
+        self.metrics: Dict[str, AgentMetrics] = defaultdict(
+            lambda: AgentMetrics(agent_name=None, openskill_model=self.openskill_model))
+
+        # For GTE
+        self.gte_game = None
+        self.gte_joint = None
+        self.gte_ratings = None
+        self.gte_marginals = None
+        self.gte_contributions_raw = None
+
+        if gte_tasks is None:
+            roles = sorted(list(set(p.role.name for g in self.games for p in g.players)))
+            self.gte_tasks = ([f'WinRate-{r}' for r in roles] +
+                              [f'KSR-{r}' for r in roles] +
+                              ['IRP', 'VSS', 'KSR'])
+        else:
+            self.gte_tasks = gte_tasks
+
+    @staticmethod
+    def _compute_elo_ratings(games: List) -> Dict[str, float]:
+        elos = defaultdict(_default_elo)
+        for game in games:
+            villager_agents = []
+            werewolf_agents = []
+            for player in game.players:
+                agent_name = player.agent.display_name
+                if player.role.team == Team.VILLAGERS:
+                    villager_agents.append(agent_name)
+                else:
+                    werewolf_agents.append(agent_name)
+            v_elos = [elos[a] for a in villager_agents]
+            w_elos = [elos[a] for a in werewolf_agents]
+            if v_elos and w_elos:
+                avg_v_elo = np.mean(v_elos)
+                avg_w_elo = np.mean(w_elos)
```

Comment:
I am concerned with just comparing average team Elos. If you pair a top-tier Werewolf with a low-tier model, their average can still imply they're a solid team. But in reality, the low-tier model can drag the team and the elo for top-tier model can suffer unfairly.

Reply:
Any alternative metrics or update rules that you recommend?